remove unnessary option setting from test runner

y-yagi opened the pull request:
If run the test over the `rake` command, because of the test patterns is passed
via `rake_run` method, do not need to be obtained from the argv.

This probably fixes #24372.

y-yagi committed:
remove unnessary option setting from test runner

If run the test over the `rake` command, because of the test patterns is passed
via `rake_run` method, do not need to be obtained from the argv.

This probably fixes #24372.

vipulnsward commented on railties/test/application/test_runner_test.rb:
```diff
@@ -502,6 +502,14 @@ def test_pass_TEST_env_on_rake_test
+
+      assert_match '1 runs, 1 assertions', output
+      assert_match 'Execute test', output
+    end
```
Can we add test from https://github.com/rails/rails/issues/24372 to accompany this?

y-yagi replied:
You mean add a test using `--rakefile` option?

If so, I've added a test.

vipulnsward commented:
Nice fix. This does work for me now: 

```
rake --rakefile "`pwd`/Rakefile" test
bundle exec rake --rakefile "`pwd`/Rakefile" test
```

But 
```
rake --rakefile "`pwd`/Rakefile" routes
```
doesn't. We can probably check and include a fix for that here.

y-yagi committed:
register rake options to `OptionParser`

In order to prevent `OptionParser::ParseError` when specify the rake options to
`routes` task.

y-yagi commented:
Thank you for pointing it out!

Also fixed `routes` task.

kaspth commented on railties/lib/rails/tasks/routes.rake:
```diff
@@ -19,6 +19,9 @@ task routes: :environment do
   OptionParser.new do |opts|
     opts.banner = "Usage: rails routes [options]"
+
+    Rake.application.standard_rake_options.each { |args| opts.on(*args) }
```
This is iffy. We'd likely need to swallow rake options for every Rails command with custom option parsing.

kaspth replied:
Hmm, come to think of it there might not be other cases since routes and test are the only rake tasks we fiddled with in Rails 5. So this feels like an ok compromise 👍

kaspth commented on railties/lib/rails/test_unit/minitest_plugin.rb:
```diff
@@ -54,7 +54,7 @@ def self.plugin_rails_options(opts, options)
     options[:color] = true
     options[:output_inline] = true
-    options[:patterns] = opts.order!
+    options[:patterns] = opts.order! unless defined?(@rake_patterns)
```
Would rather have the ivar override here, when it's defined, and remove the check down below.

kaspth replied:
Down below meaning in `plugin_rails_init` 😁

y-yagi replied:
Thanks for reviewing! I updated!

kaspth commented:
❤️

y-yagi commented:
:smile:
